addons/account/models/account_invoice.py

```diff
@@ -95,6 +95,10 @@ def _default_currency(self):
     def _get_reference_type(self):
         return [('none', _('Free Reference'))]
 
+    def _get_residual_amount_lines(self):
+        """ This function returns the aml to consider to compute the amount residual of invoices """
+        return self.sudo().move_id.line_ids
```

Comment:
since this method aims to return the lines to consider for the amount_residual computation, it should already filter on self.account_id instead of doing it at the next line